.github/workflows/code-scanning.yml

```diff
@@ -0,0 +1,64 @@
+name: "code-scanning"
+
+on:
+  push:
+  pull_request:
```

Comment:
This workflow is taking quite some time to complete. It makes more sense to switch to `on push/on pull_request` for the `master` branch only.

Reply:
Or maybe `develop` and `on: push` only, so we can catch issues before anything gets into `master`?

Reply:
The mac build seems to be very slow in this branch (which seems to be the reason for this workflow taking so long).

Reply:
> Or maybe develop and on: push only, so we can catch issues before anything gets into master?

Configuring the `on: pull-request` for `master` will also catch any issue before we merge to master, but yes, we can also add `on:push` for `develop`. The main point is to keep an sufficiently performant CI (this workflow is taking ages unfortunately)

Reply:
This may be better now.  I hit the same thing (mac build taking ~1hour).  Seemed to be caused by the initial `brew install` command, but the issue magically resolved itself when I tried to add logging to investigate further.

Reply:
Seems like the macOS build is faster now indeed. The scanning job still takes longer (~30mins) but this is much better than the 1h+ running time we had a few days back.

Reply:
I adjusted the scope of the code-scanning workflow to only run `on: push` on `develop` and `on: pull-request` on `master`. Does that sound like a right policy to you @dtebbs ? (the scanning jobs are still ~3x longer to run than the other ones so may be worth running them less frequently)

Reply:
Yes, that sounds fine.  It would be nice to have it run *before* we merge to develop, but as you say the longer run time makes that difficult. This seems like a good compromise.